cEP-0018.md

```diff
@@ -0,0 +1,177 @@
+# Integration of ANTLR into coala core
+
+| Metadata |                                               |
+| -------- | --------------------------------------------- |
+| cEP      | 18                                            |
+| Version  | 1.0                                           |
+| Title    | Integration of ANTLR into coala core          |
+| Authors  | Viresh Gupta <mailto:viresh16118@iiitd.ac.in> |
+| Status   | Proposed                                      |
+| Type     | Feature                                       |
+
+## Abstract
+
+This document describes how an API based on ANTLR will be constructed and
+maintained.
+
+## Introduction
+
+ANTLR provides parsers for various language grammars and thus we can provide
+an interface from within coala and make it available to bear writers so that
+they can write advanced linting bears. This will aim at supporting the more
+flexible visitor based method of AST traversal (as opposed to listener based
+mechanisms).
+
+The proposal is to introduce a parallel concept to the coala-bears library,
+which will be called the coala-ast hereon.
+
+## Proposed Change
+
+Here is the detailed implementation stepwise:
+
+1. There will be a separate repository named as coala-antlr which will
+   be installable via `pip install coala-antlr`.
+2. A new package would be introduced in the coala-antlr repository,
+   which would be the maintain all the dependencies for antlr-ast based
+   bears. This would be the `coantlib` package
+3. The `coantlib` package will provide endpoints relevant to the
+   visitor model of traversing the ast generated via ANTLR.
+4. Another package named `coantparsers` will be created inside
+   coala-antlr repository. This package will hold parsers generated
+   for various languages generated (with python target) beforehand for the
+   supported set of grammars.
+5. `coantlib` will have an AST loader class that will be responsible
+   for loading the AST of a given file depending on the user input, using file
+   extension as a fallback.
+   For e.g, a .c file will be loaded using the parser from
+   `coantlib` for c language.
+6. Another AST walker class will be responsible for providing endpoints for
+   traversing the AST using methods such as `get_next_node` and
+   `get_previous_node` (inside `coantlib`).
+
```

Comment:
I would like for you to add the description of where the antlr dependent bears will reside. I think they should reside in this repository.